### ISSUE TITLE
Idea: make tags optional

### ISSUE THREAD

**Issue by grahampugh** [gkluoe/git2jss#8]
I'm not really used to using tags, and have started to use them for the purposes of experimenting with git2jss. Tags are supposed to represent signficant landmarks in a project, but seem overkill for a small repo of continually updated/amended and unrelated JSS scripts.

If I'm editing and fixing a script that I need to upload to the JSS to test, and then need to amend, it means continually either creating new tags or amending an existing one, in order to be able to upload the script. 

So, tags are nice to have, but it would be cool if git2jss worked without it. If `--tag vX.Y` was omitted, it could perhaps grab the latest tag if the file is aligned with one. Or it could just grab the commit ID if no tags are present, or the file is not aligned with a particular tag.

Another optional `--version X.Y` could give people the ability to specify a version for the purposes of filling in the `@@VERSION` field without using tags.

**Comment by geoffklee** [comment]
I think I like the idea of tag being optional. I’ll need to think about the implications but since a tag is just a pointer to a specific commit ID it should be simple to implement. 

I’m slightly less keen on the idea of manually adding a version that’s independent of the tag, because then we have a ‘version’ on the JSS which isn’t necessarily ties to anything in GIT. That said, it would Be easy enough to add a command line switch to override any/all of the template variables, which could be generally useful I guess.

**Comment by geoffklee** [comment]
Thinking about this a bit more, if we don't specify a tag  I'm not sure which commit we should choose to push. The latest commit in the local repo might not have been pushed to the remote so probably isn't a good choice on its own. The latest commit in the remote, however, might not have been pulled locally, so we probably don't want to blindly push that instead. 

I guess we could try to find the latest commit that exists both on the remote and locally, but that's getting a bit hairy in terms of knowing what's going to happen when we run the tool.

I'm only really thinking here in terms of my own workflow though - what would you want/expect to happen if you didn't specify a tag? For me the value of using a tag is that I'm saying "I want this exact commit to be pushed up to the server". I could allow specifying a commit ID instead, but personally I'd much rather create a readable tag name than have to copy/paste a hex commit ID.

**Comment by geoffklee** [comment]
Hi, I think I understand the CI use case now! How about I add a —branch option which will check out and release the HEAD of that branch? It could optionally create a meaningful tag of the date/time of release. @grahampugh would that be helpful?

**Comment by grahampugh** [comment]
Hi, yes that sounds perfect! Creating a tag could also be good as long as
it was optional.

Cheers
Graham

**Comment by geoffklee** [comment]
I've implemented what I think is the most sensible way to handle this over in 
https://github.com/gkluoe/git2jss/tree/v1.1.0rc1

I'd appreciate your thoughts @grahampugh ! Doing `--branch master` will have the effect of pushing the head of master to the jss without requiring a tag. 

I've removed the tag _creation_ functionality (which was broken anyway) altogether since it's trivial to do in the shell if you do want a tag.

**Comment by grahampugh** [comment]
Hi, I'm trying to test, and probably being a bit dim, but perhaps someone else trying it would also be dim and encounter the same problems....

So, I cloned the repository, checked out branch v1.1.0rc1, and ran `sudo python setup.py install`. Note that this device already had 1.0.0 installed. 

The I attempt to run:
```
git2jss file=coreconfig-softwareupdate-run.py name="macOS Updates install script.py" --branch=master
```

I get the error that `--branch` is not recognised:

```
usage: git2jss [-i --jss-info] [-h] [ --mode MODE ] [--create]  [ --no-keychain] [--all | --file FILE [ --name NAME ] ]  --tag TAG
```

In fact, I'm not getting the new version:

```
git2jss --version 
1.0.0
```

How do I install the new version?

**Comment by geoffklee** [comment]
Thanks for testing!

Now you mention it I hadn't incremented the version number in that branch - I don't think that was causing the problem, but I've done so now.

I wonder if you inadvertently created a new local branch rather than checking out the remote one?  This works for me:

`git clone --branch v1.1.0rc1 https://github.com/gkluoe/git2jss`
`cd git2jss`
`python setup.py install`
``` 
$ git2jss --version
1.1.0rc1
```

**Comment by geoffklee** [comment]
I've released version 2.0.0 now, which includes this.
